### PR TITLE
fix(ci): handle PR base-branch retargeting via pull_request edited event

### LINKED ISSUE #94: PR workflows miss runs after retargeting base branch (omit `pull_request` type `edited`)
## Problem

When a pull request is opened against the wrong base branch (for example `main`) and later **retargeted** to `develop`, some expected GitHub Actions runs never fire. This matches GitHub’s event model, not random flakiness.

Changing the PR **base branch** emits a `pull_request` webhook with action **`edited`** (along with a `changes` payload describing the update). See [Webhook events and payloads — `pull_request`](https://docs.github.com/en/webhooks/webhook-events-and-payloads#pull_request) (`edited`: title, body, base ref, or state was edited).

For `pull_request` workflows, GitHub’s default activity types when **`types` is omitted** are only **`opened`**, **`synchronize`**, and **`reopened`** — **not** `edited`. See [Events that trigger workflows — `pull_request`](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request) (note under the activity types table).

Any workflow that either relies on that default or lists explicit `types` **without** `edited` will **not** run again solely because the base branch was changed.

### `pr-preview-environment.yml` — high impact

- Trigger: `pull_request` with `branches: [develop]` and `types: [opened, reopened, synchronize, ready_for_review, closed]`.
- **Gap:** If the PR is first opened with base `main`, this workflow does not match (`branches` filter). After retargeting to `develop`, GitHub sends **`edited`**, which is **not** in `types`, so the preview workflow **still does not run**. Previews and related PR comments never appear unless the user pushes another commit (`synchronize`) or uses **Re-run jobs** / an empty commit.

### `test.yml` — medium impact

- Trigger: `pull_request` with **`paths` filters only**; **`types` is omitted**, so GitHub uses the default (`opened`, `synchronize`, `reopened`).
- **Gap:** Retargeting the base does not trigger tests. That is misleading because the **merge ref** (`refs/pull/N/merge`) and merge commit **change** when the base changes; checks from the previous base can be **wrong or stale** relative to the new target branch.

### `check-merge-strategy.yml` — low / UX impact

- Trigger: explicit `types` without `edited`; job gated with `if: github.base_ref == 'main'`.
- **Gap (develop → main):** If someone opens against `develop` then retargets to `main`, the merge-strategy reminder may never run until a push. **Gap (main → develop):** A prior successful run when the PR targeted `main` can leave a **stale** green check on the same commit even though the PR no longer targets `main` (until the next `synchronize` or manual re-run).

## Suggested direction

1. Add **`edited`** to `pull_request.types` where re-evaluation against the current base is required (at minimum **preview** and **test** workflows).
2. Optionally narrow **`edited`** handling (job-level `if:`) so title/body-only edits do not always redeploy full preview — e.g. only run heavy jobs when `github.event.changes.base` is present on `edited` events (verify payload shape in [webhook docs](https://docs.github.com/en/webhooks/webhook-events-and-payloads#pull_request)).
3. Document for contributors: after retargeting, **push an empty commit** or **re-run failed workflows** as a workaround until the above is fixed.

## Acceptance criteria

- Retargeting an open PR from `main` to `develop` triggers PR preview workflow (or a documented, intentional skip with a clear reason).
- Retargeting re-runs **Test** when `paths` match, so checks reflect the merge against the **new** base.
- (Optional) Merge-strategy bot behavior stays correct when retargeting to/from `main` without stale or missing reminders.

### PR DESCRIPTION
**Pull request by sarah-artificerinnovations**
Closes #94

## What changed

When a PR is retargeted to a different base branch, GitHub fires a `pull_request` event with action `edited` and a `changes.base.ref.from` payload. Without `edited` in the workflow `types`, none of the PR workflows fire on retargeting — so previews never appear and tests run against a stale merge base.

### `pr-preview-environment.yml`

- Added `edited` to `types`
- Added a guard on the `preview-scope` job: skip when `edited` but `changes.base.ref.from` is empty (title/body-only edits). Only real base-branch retargets redeploy.
- All downstream jobs (`deploy-preview`, `comment`, `teardown`) are already gated on `preview-scope` so they inherit this filter.

### `test.yml`

- Added explicit `types: [opened, synchronize, reopened, edited]` (previously omitted, defaulting to the first three without `edited`)
- Added `if: github.event.action != 'edited' || github.event.changes.base.ref.from != ''` on the `tests` job so title/body-only edits don't trigger a full test run

### `check-merge-strategy.yml`

- Added `edited` to `types`
- The existing `if: github.base_ref == 'main'` already correctly routes:
  - develop → main retargeting: `edited` event, `base_ref == 'main'` → reminder runs ✅
  - main → develop retargeting: `edited` event, `base_ref == 'develop'` → job skipped ✅

## Testing

The `edited` event with `changes.base` can be verified by opening a PR against `main`, then retargeting it to `develop` via the GitHub UI and confirming the preview and test workflows appear in the Checks tab.

### CODE REVIEW

**Review by diego-artificerinnovations — APPROVED** [review]
Clean fix — the logic is correct and covers all the cases the PR description claims.

A few things I verified while reading:

**`pr-preview-environment.yml` guard:** The three-part condition is sound. For non-edited events, `github.event.action != 'edited'` short-circuits to true (no `changes` object needed). For title/body-only edits, `changes.base.ref.from` is `''` in GitHub's payload, so the job is skipped. For actual retargets, `changes.base.ref.from` is the old base name (non-empty), so it runs. GitHub Actions expression evaluation returns `''` for missing nested keys rather than throwing, so accessing `github.event.changes.base.ref.from` on a non-edited event is safe.

**`test.yml`:** The explicit `types` list intentionally omits `ready_for_review` — tests don't need to rerun when a draft is promoted, so that's the right call. The `if` guard on the `tests` job mirrors the pattern in `pr-preview-environment.yml` exactly.

**`check-merge-strategy.yml`:** The existing `if: github.base_ref == 'main'` already handles the retarget-from-main edge case correctly (job skips when base is `develop`). No additional guard needed.

**One thing worth a quick manual verify:** the `concurrency` group on `pr-preview-environment.yml` is keyed to `github.event.pull_request.number`. On a retarget `edited` event this is still the PR number, so any in-flight deploy for the old base would be cancelled correctly. That's the right behavior, but worth confirming it cancels-and-redeploys (not just cancels) when the base changes.

Approved — happy for Brad to merge.

**Review by rahul-artificerinnovations — APPROVED** [review]
Clean, targeted fix. The logic is correct and the guard pattern is the right call.

**`check-merge-strategy.yml`** — `edited` added, no guard. Fine: this job re-checks PR metadata (base branch, merge strategy), so re-running on any edit is harmless.

**`pr-preview-environment.yml`** — guard expression:

```yaml
(github.event.action != 'edited' || github.event.changes.base.ref.from != '')
```

Correct. `github.event.changes.base.ref.from` is absent (evaluates to `''` in GitHub Actions expressions) for title/body edits, so the OR short-circuits to false and the job skips. It is non-empty only when the base was retargeted, which is exactly when redeploy is needed. ✓

**`test.yml`** — same guard, same analysis, same result. ✓

**One coordination note (non-blocking):** PR #90 (Diego — signed cookies) also touches `pr-preview-environment.yml` and adds the same `edited` event and the same guard expression. Both target `develop`, so the second to merge will see a conflict in that file. No action needed on your side — just flagging so the merge order gets coordinated.

Approving.

**Review by mei-artificerinnovations — APPROVED** [review]
Clean and correct. The three-way guard is the right design:

```
(github.event.action != 'edited' || github.event.changes.base.ref.from != '')
```

When a PR is edited with only a title/body change, `changes.base.ref` is absent and evaluates to empty string in Actions — condition gates it out. When the base branch is retargeted, `changes.base.ref.from` is the old base name — non-empty, condition passes, deploy fires. Title/body edits stay quiet.

`check-merge-strategy.yml` didn't need the extra condition because the existing `github.base_ref == 'main'` already handles direction correctly for both retarget directions.

One thing to verify manually: `edited` fires on label changes too. Labels don't populate `changes.base.ref`, so they're filtered by the same condition — no false triggers. ✓

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -39,6 +40,8 @@ permissions:
 
 jobs:
   tests:
+    # On edited events, only run when the base branch was retargeted (not title/body-only edits).
+    if: github.event.action != 'edited' || github.event.changes.base.ref.from != ''
```
[the remark this answers is not in the document]
Comment (sarah-artificerinnovations) [reply .github/workflows/test.yml]: Good catch. Switched to `github.event.changes.base` (truthy object check) in `c6184f3`. When the base branch is retargeted `changes.base` is a JSON object (truthy); when it is a title/body edit `changes.base` is absent (empty string — falsy). This avoids any ambiguity around how GHA handles deeply-nested missing-path access.

File: .github/workflows/pr-preview-environment.yml [hunk 2]
```diff
@@ -23,8 +24,12 @@ concurrency:
...
 jobs:
   preview-scope:
-    # Skip main→develop sync PRs (no feature preview; diff would match everything on develop).
-    if: github.event.action != 'closed' && (github.base_ref != 'develop' || github.head_ref != 'main')
+    # Skip main→develop sync PRs, closed events not followed by teardown, and edited events
+    # that only changed the PR title/body (not the base branch — those should redeploy).
+    if: >-
+      github.event.action != 'closed' &&
+      (github.base_ref != 'develop' || github.head_ref != 'main') &&
+      (github.event.action != 'edited' || github.event.changes.base.ref.from != '')
```
[the remark this answers is not in the document]
Comment (sarah-artificerinnovations) [reply .github/workflows/pr-preview-environment.yml]: Fixed in `c6184f3` — same change as in `test.yml`: replaced `github.event.changes.base.ref.from != ''` with `github.event.changes.base` (truthy object check) on line 32 of `preview-scope`'s `if` condition.

File: .github/workflows/check-merge-strategy.yml [hunk 3]
```diff
@@ -2,7 +2,7 @@ name: Check Merge Strategy
 
 on:
   pull_request:
-    types: [opened, synchronize, reopened, ready_for_review]
+    types: [opened, synchronize, reopened, ready_for_review, edited]
```
[the remark this answers is not in the document]
Comment (sarah-artificerinnovations) [reply .github/workflows/check-merge-strategy.yml:6]: Fixed in `c6184f3`. Added `&& (github.event.action != 'edited' || github.event.changes.base)` to the job-level `if` in `check-merge-strategy.yml`. This skips the workflow entirely on title/body-only edits, avoiding the unnecessary comment churn.
